### DISCUSSION TITLE
Zigbee2mqtt: bloccare accoppiamenti router/end devices

### DISCUSSION THREAD

**Discussion by faustozago** [indomus/forum#2566] [ZigBee, Z-Wave, Thread, Matter]
Ciao a tutti,
mi succede spesso che , dopo aver accoppiato accuratamente gli end devices a determinati router/coordinator (a mia scelta, a volte piu' lontani, ma con sufficiente ricezione),  dopo qualche giorno me li ritrovo accoppiati ad altri router.
Esiste un modo per bloccare l'accoppiamento dei dispositivi?
In particolare mi servirebbe che 4 dispositivi rimangano accoppiati al coordinator , per gli altri non è un problema dove si accoppiano.
Ho visto che nel menu  Impostazioni->Principale , c'e' una casella PERMIT JOIN che si puo' spuntare....a cosa serve questa funzione?

Grazie

**Comment by snakuzzo** [comment]
No...non puoi forzare niente. E' tutto gestito in automatico senza possibilità di intervento. Non deve essere un tuo problema...questo è proprio uno dei punti di forza del protocollo zigbee. Perché ti fai sto problema ? Che necessità hai ?
Se scolleghi il router a cui si sono collegati, in automatico si ricollegheranno in automatico al coordinator e/o ad uno dei router della tua rete.

**Comment by faustozago** [comment]
Ho necessita' perche' vorrei avere i componenti del sistema antifurto tutti a batteria : 
- i sensori di movimento/porte/finestre/sirena  sono end point gia' a batteria .
- il coordinator USB è sulla porta del raspberry che è sotto UPS .
- 
Se i sensori rimangono collegati al coordinator funziona tutto, ma se si staccano dal coordinator e si collegano ad un router a 230V , quando salta la corrente questi non funzionano.

Sinceramente non ho provato a vedere se quando salta la corrente (quindi i router a 230V tutti fuori uso) , i vari dispositivi a batteria si connettono automaticamente al coordinator ; per funzionare pero' dovrebbero connettersi in tempi molto brevi......

**Reply by snakuzzo** [reply]
Ti ho già detto che è tutto gestito in automatico e si collegheranno in automatico al coordinator, se è nel range. Se non è nel range hai perso a prescindere.
E comunque, ripeto, NON PUOI forzare la connessione ad uno specifico device ( che sia coordinator o router ) ... è gestito tutto in automatico dal protocollo zigbee. Fine.

**Reply by faustozago** [reply]
OK tutto chiaro , grazie

**Comment by faustozago** [comment]
Senza aprire un'altro post, faccio qui un'altra domanda relativa sempre al mio problema:
è possibile avere 2 istanze Z2M nello stesso account H.A. ?  Naturalmente ogni istanza avra' il suo coordinator.
In questo modo potrei mettere in una dispositivi che voglio siano sempre associati al coordinator "A" , senza mettere altri router  ; nell'altra metto tutto il resto ,con un coordinator "B" +vari router , e i dispositivi che non mi interessa dove si collegano.

Oppure potrei fare rete Z2M  e una rete ZHA  , configurate come sopra (una con solo coordinator,e dispositivi a batteria, senza altri router ; l'altra con tutto il resto).

Grazie

**Reply by snakuzzo** [reply]
Non è questione di account di HA, forse intendevi istanza.
Comunque sì ne puoi usare n di reti zigbee

**Reply by faustozago** [reply]
Sai consigliarmi qualche guida per fare questo?  E' sufficiente installare di nuovo l'addon di Z2M ?  Prende in automatico un nome diverso?
Grazie

**Reply by snakuzzo** [reply]
Non ci sono guide. Non penso si possa fare utilizzando addon (o comunque non con le classiche installazioni su HA OS).
Per rendertela facile, su HA OS potresti fare così, se hai bisogno solo di un'altra rete.

- continui ad usare z2m così come stai facendo
- e poi colleghi un altro device coordinator e lo usi con ZHA (integrazione che trovi su core)
